spec/index.md

```diff
@@ -670,19 +678,25 @@ See [The system state tree](#state-tree) for details on the state tree.
 
 ### Request: Query call {#http-query}
 
-A query call is a fast, but less secure way to call a canister. Only methods that are explicitly marked as "query methods" by the canister can be called this way.
+A query call is a fast, but less secure way to call a canister. Only methods that are explicitly marked as "query methods" and "composite query methods" by the canister can be called this way. In contrast to a query method, a composite query method can make further calls to query and composite query methods of canisters on the same subnet.
```

Comment:
```suggestion
A query call is a fast, but less secure way to call a canister. Only methods that are explicitly marked as "query methods" and "composite query methods" by the canister can be called this way. In contrast to a query method, a composite query method can make further calls to query and composite query methods of canisters on the same subnet. The maximum depth of a composite query's call tree may, however, be limited by the implementation.
```

Reply:
There is one more restriction of this form: the total amount of cycles spent on all the message executions while evaluating a composite query call is bounded. I believe we should either mention that both or none. What would you prefer?

Reply:
I think at this point we should (at least informally) indicate any restrictions that might be interesting for users. Actually, how about regular query methods? I know that there's also a limit on how long a replica will execute these for, but I don't know if the limit is expressed in terms of cycles, or if it's just a timer? If we want to move to replicated queries, I suppose we will need a limit based on cycles. 

Maybe for the time being we can just be vague and say that the implementation can impose an arbitrary limit on the execution time of both query and composite query methods?

Reply:
The implementation indeed imposes this restriction:
```
/// The maximum time in seconds a query call is allowed to run.
pub(crate) const MAX_TIME_PER_COMPOSITE_QUERY_CALL: Duration = Duration::from_secs(10);
```

So I've added all the three restrictions to the spec: 1680a26fed8eb9c2d77e8a82d8330611bbbf1c36

Reply:
Thanks!